homeassistant/components/deconz/diagnostics.py

```diff
@@ -0,0 +1,39 @@
+"""Diagnostics support for deCONZ."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.system_info import async_get_system_info
+
+from .gateway import get_gateway_from_config_entry
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, config_entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    gateway = get_gateway_from_config_entry(hass, config_entry)
+    diag: dict[str, Any] = {}
+
+    diag["home_assistant"] = await async_get_system_info(hass)
+    diag["config_entry"] = dict(config_entry.data)
```

Comment:
Please redact sensitiv data from the output.
There is a helper fort that -> [`homeassistant.components.diagnostics.async_redact_data()`](https://github.com/home-assistant/core/blob/511546631a5b240c5ef4c0dd25c4a542edc62a72/homeassistant/components/diagnostics/util.py#L13)

Reply:
Fixed https://github.com/home-assistant/core/pull/64754